### ISSUE TITLE
Client crash when accepting party invitation

### ISSUE THREAD

**Issue by loentar** [Darkin47/SagaRevised#17]
Client crashes when accepting party invitation and character levels > 1. When both characters are lvl1 clients does not crash.

```
OS: Windows XP 5.1 (Build: 2600)
CPU: GenuineIntel PentiumPro-class processor @ 10 MHz with 2047MB RAM
Video: Intel(R) HD Graphics 4600 (4352)

General protection fault!

History: Rag2NetRecvMsg::RecvPartyInfoNew <- Rag2NetRecvMsg::RecvPartyInfo <- Rag2NetRecvMsg::RecvParty <- Rag2NetRecvMsg::ReceivedMsg <- Rag2NetRecvMsg::ReceivedGACT <- Rag2NetRecvMsg::ReceivedPacket <- CLoginConnection::MessageProcess <- CNetConnector::DoRead <- CNetClientSock::Update <- CNetMessageMgr::Tick <- UGameEngine::Rag2Tick <- ClientTick-RequestSection <- ClientTick <- UGameEngine::Tick <- UpdateWorld <- MainLoop
```

**Comment by Daniel4rt** [comment]
This is the source of Ragnarok Client about that.
[Rag2NetRecvMsgParty.h.txt](https://github.com/JulienGrv/saga-revised/files/1326967/Rag2NetRecvMsgParty.h.txt)
[Rag2NetRecvMsgParty.cpp.txt](https://github.com/JulienGrv/saga-revised/files/1326968/Rag2NetRecvMsgParty.cpp.txt)

And party creation:
https://github.com/JulienGrv/saga-revised/blob/master/Saga.Map/Netwok/Client/Client%20Party.cs

**Comment by Daniel4rt** [comment]
Friend Register is there:
https://github.com/JulienGrv/saga-revised/blob/master/Saga.Map/Netwok/Client/Client%20Friendlist.cs#L15

**Comment by loentar** [comment]
Yes, I already examined client sources. Structures looks ok, data at first sight - ok too.. Do someone tried to build client (in debug mode), and it's possible at all?

Looks like client crash when other's party char level is higher than 7-8, does not crash when it 6 or lower (the difference between levels seems does not matter).

The strange thing, I commented out the whole CM_PARTYINVITATIONACCEPT method, but invitation is still work!? maybe it working from another piece of code? (I sure I recompiled the sources)

**Comment by JulienGrv** [comment]
> The strange thing, I commented out the whole CM_PARTYINVITATIONACCEPT method, but invitation is still work!? maybe it working from another piece of code? (I sure I recompiled the sources)

This may work on client side but on server side I suppose nothing has been done (nothing saved in database for example).

**Comment by loentar** [comment]
> The strange thing, I commented out the whole CM_PARTYINVITATIONACCEPT method, but invitation is still work!? maybe it working from another piece of code? (I sure I recompiled the sources)

My fault.. I had 2 map server started.. 


Further research:
Packet log when client(s) do not crash:

```
Recieved: 0E03 data 0F00060000000501000005000E0301
PARTY PACKET Recieved: 0E03 data 0F00060000000501000005000E0301
Packet Sent: 0D02 data 0F00070000000601000005000D0200
Packet Sent: 0D04 data 290107000000060100001F010D040107000000010100000000000107000000640065006D0000000000000000000000000000000000000000000000000000000000680003FA00FA006E006E000101010100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001
Packet Sent: 0D05 data 3A00070000000601000030000D0501060000000153006800690072006F0000000000000000000000000000000000000000000000000000000000
Packet Sent: 0D04 data 290106000000060100001F010D040107000000010100000000000107000000640065006D0000000000000000000000000000000000000000000000000000000000680003FA00FA006E006E0001010101010600000053006800690072006F00000000000000000000000000000000000000000000000000680003FA00FA006E006E000101010100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001
Packet Sent: 0D06 data 2200060000000601000018000D060107000000680003FA00FA006E006E0001010101
Packet Sent: 0D06 data 2200070000000601000018000D060106000000680003FA00FA006E006E0001010101
Packet Sent: 0D15 data 1B00060000000601000011000D15010700000090CD2301C49F3700
Packet Sent: 0D15 data 1B00070000000601000011000D1501060000005CD22501C63A3B00
```


When it crash (after few levels for both chars):

```
Recieved: 0E03 data 0F00010000000501000005000E0301
PARTY PACKET Recieved: 0E03 data 0F00010000000501000005000E0301
Packet Sent: 0D02 data 0F00020000000601000005000D0200
Packet Sent: 0D04 data 290102000000060100001F010D040102000000010100000000000102000000640065006D0000000000000000000000000000000000000000000000000000000000680003C602C6026E006E000709010900000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001
Packet Sent: 0D05 data 3A00020000000601000030000D0501010000000153006800690072006F0000000000000000000000000000000000000000000000000000000000
Packet Sent: 0D04 data 290101000000060100001F010D040102000000010100000000000102000000640065006D0000000000000000000000000000000000000000000000000000000000680003C602C6026E006E0007090109010100000053006800690072006F00000000000000000000000000000000000000000000000000680003C201C2016E006E000706010600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001
Packet Sent: 0D06 data 2200010000000601000018000D060102000000680003C602C6026E006E0007090109
Packet Sent: 0D06 data 2200020000000601000018000D060101000000680003C201C2016E006E0007060106
Packet Sent: 0D15 data 1B00010000000601000011000D15010200000090CD2301C49F3700
Packet Sent: 0D15 data 1B00020000000601000011000D1501010000005CD22501C63A3B00
```

**Comment by loentar** [comment]
Crash occurs when received LP is more than 5. Looks like it used for display-only and causes out-of-index -like crash.

**Comment by JulienGrv** [comment]
Fixed with PR #18
